### ISSUE TITLE
[NEXT-1125] Misleading deopted-into-client-rendering warning on build

### ISSUE THREAD

**Issue by sophiebits** [vercel/next.js#48442]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 19.5.0
      npm: 9.3.1
      Yarn: 1.22.19
      pnpm: 8.2.0
    Relevant packages:
      next: 13.3.1-canary.8
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-f51au5 (you may have to download locally; I couldn't get next build to run in stackblitz)

### To Reproduce

If the root page or layout calls headers() or cookies() outside of any Suspense boundary, then `next build` outputs this warning for each affected page:

```
warn  - Entire page / deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /
```

If it's the root layout that uses that dynamic data, it prints this for every single route.

While it's true that the page can't be statically generated, it's not true that it all deopted into client-side rendering. The page does SSR just fine.

### Describe the Bug

Error message is misleading.

### Expected Behavior

The error should not appear, or should have different text.

<sub>[NEXT-1125](https://linear.app/vercel/issue/NEXT-1125/misleading-deopted-into-client-rendering-warning-on-build)</sub>

**Comment by HamAndRock** [comment]
Having the same issues after updating to 13.3/canary. https://github.com/vercel/next.js/issues/48404 this is probably the root cause if you are also using route groups.

**Comment by Nickman87** [comment]
Same issue here, all my pages (inside of groups) give me this same error/warning and 0B size.
I don't use the `useSearchParams` hook anywhere

**Comment by RyanGaudion** [comment]
Same issue here. I think it's due to having a "use client" component in my root layout?

**Comment by Nickman87** [comment]
@RyanGaudion I think in that case it's normal that you deopt into client rendering yes... :)

**Comment by RyanGaudion** [comment]
@Nickman87 my understanding of this from [here](https://beta.nextjs.org/docs/rendering/server-and-client-components) was that client components were "prerendered on the server and hydrated on the client" and hence the page should still be able to be pre-rendered on server and only add interactivity to my cookie banner (client component referenced in the root layout) on client side" however I get this warning:

`Entire page /blog/[slug] deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /blog/[slug]` 

Sorry for my misunderstanding but should this be the expected behaviour that the whole page is no longer able to be pre-rendered due to a single client component that is used in the root layout?

**Comment by HamAndRock** [comment]
@RyanGaudion root layout can't be a client component that's in docs, hovewer if you are getting this error and you are not using useSearchParams anywhere or they are wrapped in Suspense/Dynamic Rendering this should not be happening. It started doing this on 13.3 try using 13.2 to see if it's still there. I think the root cause here are the route groups

**Comment by keegandonley** [comment]
I'm seeing this warning as well, but only on routes that don't use the `experimental-edge` runtime. My few routes that use that don't emit the waning unless I change the runtime back to default.

**Comment by borispoehland** [comment]
> Having the same issues after updating to 13.3/canary. #48404 this is probably the root cause if you are also using route groups.

I can confirm this. The route group causes the warning to appear

**Comment by RyanGaudion** [comment]
Issue resolved for me by wrapping my analytics component (in the root layout) in a [suspense boundary](https://beta.nextjs.org/docs/data-fetching/streaming-and-suspense)

**Comment by zartinn** [comment]
I have the same problem as @RyanGaudion 
I think this warning is not correct as it even appears after I commented out all my client side components in the RootLayout and and page that it was given me the warning.

And when if I leave my client side components in my layout and my page. My pages are still rendered on the server looking into the HTML that is coming back. Something is definitely odd here with the warning.

**Comment by Arctomachine** [comment]
> Issue resolved for me by wrapping my analytics component (in the root layout) in a [suspense boundary](https://beta.nextjs.org/docs/data-fetching/streaming-and-suspense)

In my case the cause of problem was client side component for parsing analytic params from url. The component was called in root layout. After adding it I got this warning for every page. After wrapping it into suspense everything is fixed.
The silly thing about it is that component could not possibly prevent pages from generating as the only thing it returns as jsx is always null.

**Comment by DenisBessa** [comment]
In my case, adding a [loading.tsx file](https://beta.nextjs.org/docs/routing/loading-ui) to the root of the /app folder solved the problem.

**Comment by KingXP-Pythoner** [comment]
I immediately started experiencing this client-side deopting warning after I upgraded to v13.3 and higher. I wasn't using useSearchParams and because I had no errors before the upgrade, I was quite confused what the issue was. I attempted all the solutions including suspense boundaries and removing the use of cookies(). Only solution that worked was to export dynamic=force-dynamic in my /dashboard layout.tsx to force dynamic rendering. I was not satisfied with the fact I had to resort to this solution so after tracking my deployments, I noticed the major change I made was the version upgrade. Once I downgraded to 13.2, the warnings disappeared. Also note that I was using route groups in v13.2 as well as v13.3 and above.

**Comment by red2678** [comment]
Still having this issue on 13.4

**Comment by boatilus** [comment]
In my case, I also use route groups and noticed this behavior in 13.4. After opting all routes into the Edge runtime (via exporting `runtime = 'edge'`), I no longer see this warning.

**Comment by arclogos132** [comment]
My case was using supabase that used headers() when creating the client.

**Comment by F-Moody** [comment]
`import '../../styles/globals.css'
import MessageToast from '../(shared)/(components)/messageToast'
import Head from './home/head'
import { getServerSession } from 'next-auth'
import { authOptions } from '../../pages/api/auth/[...nextauth]'
import PageContextProvider from './home/(components)/pageContextProvider'
import Logo from '../(shared)/(components)/Logo'
import ProfileDropDown from '../(shared)/(components)/profileDropDown'
import Navbar from '../(shared)/(components)/navbar'

export default async function RootLayout({ children }) {
  const session = await getServerSession(authOptions)

  return (
    <html>
      <Head />
      <body>
        <div className="flex flex-col h-full">
          <div className="flex py-4 px-10 justify-between mobile:hidden">
            <Logo />
            {<ProfileDropDown user={session?.user} />}
          </div>
          <div className="grid grid-cols-6 h-full large:flex large:flex-col">
            <div className="px-6 py-10 col-span-1 flex flex-col gap-2 large:w-full large:h-[52px] large:p-0">
              <Navbar isLoggedUser={!!session?.user} />
            </div>
            <div className="col-span-5 mobile:flex mobile:flex-col">
              <PageContextProvider session={session}>
                <MessageToast />
                {children}
              </PageContextProvider>
            </div>
          </div>
        </div>
      </body>
    </html>
  )
}`

I started to have the same problem in every route.
This is an example of a Layout component. 

I also tried to wrap the `PageContextProvider` in a `Suspense` but nothing.
This issue wasn't here before upgrading.

**Comment by polnikale** [comment]
I have this warning when I have a "Link" component inside my static page. And when I hover it - I see the warning. I mean it's expected, right?

When I set prefetch={false} to the link - the warning is gone

**Comment by mkimbo** [comment]
I was experiencing the same and solved it by wrapping my client components in Root Layout with Suspense. I'm on v13.4

**Comment by yersultanur** [comment]
`import { Suspense } from "react"
import Link from "next/link"
import { getDictionary } from "@/get-dictionary"
import { Locale } from "@/i18n-config"

import getMarketingConfig from "@/config/marketing"
import { getCurrentUser } from "@/lib/session"
import { cn } from "@/lib/utils"
import { MainNav } from "@/app/[lang]/components/main-nav"
import { SiteFooter } from "@/app/[lang]/components/site-footer"
import LocaleSwitcher from "@/app/[lang]/components/switcher"
import { buttonVariants } from "@/app/[lang]/components/ui/button"
import UserAccountNav from "@/app/[lang]/components/user-account-nav"

interface MarketingLayoutProps {
  children: React.ReactNode
  params: {
    lang: Locale
  }
}

export default async function MarketingLayout({
  children,
  params,
}: MarketingLayoutProps) {
  const user = await getCurrentUser()
  const dict = await getDictionary(params.lang as Locale)
  const marketingConfig = await getMarketingConfig({
    params: { lang: params.lang },
  })
  return (
    <div className="flex min-h-screen flex-col">
      <header className=" sticky top-0 z-40 border-b bg-background ">
        <div className="container relative flex h-16 items-center justify-between py-6">
          <MainNav items={marketingConfig.mainNav} lang={params.lang} />

          <span className="absolute right-32 flex">
            <LocaleSwitcher />
          </span>
          <nav>
            <Suspense fallback={<p>Загрузка личных данных...</p>}>
              {user ? (
                <UserAccountNav
                  user={{
                    firstName: user.firstName,
                    email: user.email,
                  }}
                  dict={dict.UserAccountNav}
                  lang={params.lang}
                />
              ) : (
                <Link
                  href={`/${params.lang}/login`}
                  className={cn(
                    buttonVariants({ variant: "secondary", size: "sm" }),
                    "px-4"
                  )}
                >
                  {dict.nav.login}
                </Link>
              )}
            </Suspense>
          </nav>
        </div>
      </header>
      <main className="flex-1">{children}</main>
      <SiteFooter dict={dict} />
    </div>
  )
}
`

Here is a problem, when I have await getCurrentUser issue stays no matter what, I don't know how I can wrap it to suspense

**Comment by timneutkens** [comment]
Tracked this down to an accidental conflict between two error digests. Fixed it in #50219. It would specifically incorrectly show up when calling cookies / headers and such.
